lib/protocol/http/header/etags.rb

```diff
@@ -8,15 +8,32 @@
 module Protocol
 	module HTTP
 		module Header
-			# This implementation is not strictly correct according to the RFC-specified format.
 			class ETags < Split
 				def wildcard?
 					self.include?('*')
 				end
 				
+				# This implementation is not strictly correct according to the RFC-specified format.
 				def match?(etag)
 					wildcard? || self.include?(etag)
 				end
+				
+				# Useful with If-Match
+				def strong_match?(etag)
+					wildcard? || self.reject{|t| weak_tag?(t)}.include?(etag)
+				end
+				
+				# Useful with If-None-Match
+				def weak_match?(etag)
+					opposite_tag = weak_tag?(etag) ? etag[2..-1] : "W/#{etag}"
```

Comment:
Can we move `opposite_tag` into a private method?

Reply:
Sure thing